packages/components/src/templates/next/components/internal/Navbar/NavItem.tsx

```diff
@@ -70,61 +70,61 @@ export const NavItem = forwardRef<HTMLButtonElement, NavbarItemProps>(
           <BiChevronDown className={chevron({ isOpen })} />
         </button>
         {isOpen && (
-          <FocusScope contain restoreFocus autoFocus>
-            <div
-              className="fixed inset-0"
-              style={{
-                top,
-              }}
-            >
-              <div ref={megaMenuRef} className={megamenu()}>
-                <div className="mx-auto flex w-full max-w-screen-xl flex-col gap-8 px-10 pb-16 pt-12">
-                  <div className="flex w-full flex-row items-start">
-                    <div className="flex flex-col gap-1">
-                      <h1 className="prose-display-sm text-base-content">
-                        {name}
-                      </h1>
-                      {description && (
-                        <p className="prose-label-sm-regular text-base-content-subtle">
-                          {description}
-                        </p>
-                      )}
-                    </div>
-
-                    {/* Spacer */}
-                    <div className="flex-1" />
-
-                    <button
-                      onClick={onCloseMegamenu}
-                      aria-label="Close navigation item"
-                      className="flex h-[2.125rem] w-[2.125rem] items-center justify-center text-[1.5rem]"
-                    >
-                      <BiX />
-                    </button>
+          <Dialog
+            open={isOpen}
+            onClose={onCloseMegamenu}
+            className="fixed inset-0 bg-canvas-overlay/40"
```

Comment:
issue(non-blocking): Clicking on the nav item to close it doesn't actually close it.

https://github.com/user-attachments/assets/9b876119-3e0e-492a-90f3-cbc0c0533ba9

But okay we can settle in a separate PR.